### PR TITLE
#143412449 JWT for Signin

### PR DESCRIPTION
**Pull request by LWanjiru**
#### What does this PR do?
- It adds a JWT token upon successful signin.
 #### Description of Task to be completed?
- Users once authenticated via the login api should be returned a JWT to be required on subsequent call to all other endpoints that require authentication. 
 #### How should this be manually tested?
- By using the credentials(email and password) on Postman after running the application.
 #### What are the relevant pivotal tracker stories?
- #143412449

### CODE REVIEW

**Comment by gngchrs** [comment]
I think this PR should have postman screenshots

File: app/controllers/users.js [hunk 1]
```diff
@@ -1,65 +1,61 @@
-/**
- * Module dependencies.
- */
-var mongoose = require('mongoose'),
-  User = mongoose.model('User');
-var avatars = require('./avatars').all();
+/* global next:true*/
+/* eslint no-undef: "error"*/
+/* eslint no-underscore-dangle: ["error", { "allow": ["_id"]}]*/
+/* eslint no-unused-vars: ["error", { "argsIgnorePattern": "^_" }]*/
```
Comment (gngchrs) [inline app/controllers/users.js]: minor: Why are we overwriting linting rules?

File: app/controllers/users.js [hunk 2]
```diff
@@ -152,37 +142,52 @@ exports.addDonation = function(req, res) {
...
       next();
     });
-};
\ No newline at end of file
+};
+
+ // Attach token to user credentials after authentication
+exports.loginWithEmail = (req, res) => {
+  User
+    .findOne({ email: req.body.email })
```
Comment (gngchrs) [inline app/controllers/users.js]: There should be better error handling here. What if email or password are empty. We should not respond with `user not found`. Or try to authenticate an empty pasword

File: package.json [hunk 3]
```diff
@@ -42,6 +42,8 @@
     "gulp-nodemon": "^2.2.1",
     "gulp-sass": "^3.1.0",
     "jade": "~1.11.0",
+    "jsonwebtoken": "^7.3.0",
+    "jwt-simple": "^0.5.1",
```
Comment (gngchrs) [inline package.json]: I didn't see this used anywhere

File: package.json [hunk 4]
```diff
@@ -52,6 +54,7 @@
     "passport-facebook": "~2.1.1",
     "passport-github": "~1.1.0",
     "passport-google-oauth": "~1.0.0",
+    "passport-jwt": "^2.2.1",
```
Comment (gngchrs) [inline package.json]: ditto

File: app/controllers/users.js (hunk 1 above)
Comment (LWanjiru) [reply app/controllers/users.js]: The _id and 'next' variables are in use in different files that we are not working on currently and touching them crashes the application, making it hard for us to test the functionality of the feature.

File: app/controllers/users.js (hunk 2 above)
Comment (LWanjiru) [reply app/controllers/users.js]: We will implement that.

File: package.json (hunk 3 above)
Comment (LWanjiru) [reply package.json]: We were using the "jsonwebtoken" to work on the signin feature. Removed the 'jwt-simple' from the dependencies.

File: package.json (hunk 4 above)
Comment (LWanjiru) [reply package.json]: Done.

**Pull request closed without merging**